### PR TITLE
chore: update the contributing guide for post v1 maintenance mode

### LINKED ISSUE #20041: CONTRIBUTING.md needs some minor updates in Getting Started / Pack
### Describe the issue

Below is a long-winded way of saying:

# Short version

## What did you try?
Read CONTRIBUTING.md and follow its instructions for packaging a module in aws-cdk

## What happened?
I was eventually able to run `yarn run package` but did not find any python packages being produced

## What behavior did you expect:
To understand how to locally work in the repo then produce packages that I can use for local testing/development and (most importantly to me) understand how to do this for *v2* of this lib.

# Details

I'm trying to get ready to contribute some updates to this project but, before I dive in, I wanted to get familiar and get an idea of how I can try out local changes and package them to see if they will do what I need in a Python based app. From reading *CONTRIBUTING.md*, my understanding of how best to go from zero to python-package would be more or less (after installing node/yarn/.NET/python/docker:
```bash
$ git clone https://github.com/{your-account}/aws-cdk.git
$ cd aws-cdk
$ yarn install
$ cd packages/@aws-cdk/aws-ec2
$ ../../../scripts/buildup  # Unclear to me if this is necessary but it doesn't hurt
$ cd <root-of-cdk-repo>
$ docker run --rm --net=host -it -v $PWD:$PWD -w $PWD jsii/superchain
docker$ cd packages/@aws-cdk/aws-ec2
docker$ ../../../scripts/foreach.sh --up yarn run package
```

That last step is where things go wonky. I spent a few days off and on trying to get that to work and here is the short version of what I found:
Running things as described reliably produces an error for me:
```
aws-cdk/packages/aws-cdk: yarn run package  (12 remaining)
yarn run v1.22.5
$ cdk-package
Validating circular imports
Error: spawnSync .../aws-cdk/node_modules/esbuild-linux-64/bin/esbuild ENOBUFS
Package failed. Total time (2.5s)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error: last command failed. fix problem and resume by executing: .../dev/aws-cdk/packages/@aws-cdk/aws-ec2/../../../scripts/foreach.sh
directory:    .../aws-cdk/packages/aws-cdk
```
After a little digging, I found that the `:latest` version of the jsii docker referenced in *CONTRIBUTING.md* uses node10, while the docs say that >=14.15.0 is required. If I instead use `:1-buster-slim-node14`, that gets me further, but another change between versions is that `:latest` runs as `root`, while the latter does not, so it runs into a number of file permissions issues trying to operate on folders/files that the docker user does not own. The most reliable way I was able to get past that was to force the user back to `root` with the `--user root` arg for `docker run` (not ideal, obviously). So, changing the package part of the process, gets me a successful package run:
```bash
$ docker run --rm --net=host -it --user root -v $PWD:$PWD -w $PWD jsii/superchain:1-buster-slim-node14
docker$ cd packages/@aws-cdk/aws-ec2
docker$ ../../../scripts/foreach.sh --up yarn run package
```
*But wait, there's more!*

Looking at the command output, I was excited to go find my python packages that I could theoretically go test in a demo app, but to my surprise, there was `js`, `dotnet`, and `java` output in all the relevant `dist/` folders, but nothing for `python`. Monitoring  the `/tmp/foreach.stdio` file during another attempt showed me that it did 'finish' the python packaging, though I can find no artifacts of that having happened :confused: .

So that leaves me with two problems:
1) I can't seem to produce a package for python in the module used as an example for CONTRIBUTING.md
2) Once I manage that, I'm not sure how to go about doing the same for cdk v2

### Links

[CONTRIBUTING.md](../blob/186dc329aa87157d11e3d0148c1d9d957adcf1fa/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by madeline-k**
Now that v1 is in maintenance mode, we need some updates to the contributing guide. 

The biggest change to the developer workflow now that v1 is in maintenance mode, is that the main development branch, `main`, has the v2 source code. This does not have many practical impacts on developer workflow, because in most cases we will continue to just iterate on a single service module in its `packages/@aws-cdk/aws-<service>` directory.

This PR adds instructions for building and testing `aws-cdk-lib` and the individual `-alpha` packages, which developers might need to do occasionally. 

closes https://github.com/cdklabs/cdk-ops/issues/1933 

closes #20041 

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -863,6 +892,69 @@ run as part of the regular build, since they have some particular requirements.
...
+main differences between the alpha package is naming of the package, and import
+statements.
+
+First, make sure the following packages are built:
+  - packages/@aws-cdk/assert
+  - packages/aws-cdk-lib
+  - tools/individual-pkg-gen
+
+The following command will create all of the alpha packages by copying files
+from their source directories under `packages/@aws-cdk/aws-<service>, and it
```
Comment (kaizencc) [inline CONTRIBUTING.md:942]:
```suggestion
from their source directories under `packages/@aws-cdk/aws-<service>`, and it
```

**Review by kaizencc — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,34 +13,59 @@ Any code you submit will be released under that license.
...
+      - [I added a dependency but it's not being picked up by the build](#i-added-a-dependency-but-its-not-being-picked-up-by-the-build)
+      - [I added a dependency but it's not being picked up by a `watch` background compilation run.](#i-added-a-dependency-but-its-not-being-picked-up-by-a-watch-background-compilation-run)
+      - [I added a dependency but it's not being picked up by Visual Studio Code (I still get red underlines).](#i-added-a-dependency-but-its-not-being-picked-up-by-visual-studio-code-i-still-get-red-underlines)
+      - [I'm doing refactorings between packages and compile times are killing me/I need to switch between differently-verionsed branches a lot and rebuilds because of version errors are taking too long.](#im-doing-refactorings-between-packages-and-compile-times-are-killing-mei-need-to-switch-between-differently-verionsed-branches-a-lot-and-rebuilds-because-of-version-errors-are-taking-too-long)
+  - [Debugging](#debugging)
+    - [Connecting the VS Code Debugger](#connecting-the-vs-code-debugger)
+    - [Run a CDK unit test in the debugger](#run-a-cdk-unit-test-in-the-debugger)
+  - [Related Repositories](#related-repositories)
```
[the remark this answers is not in the document]
Comment (kaizencc) [reply CONTRIBUTING.md:68]: AFAICT (can tell) from the diff this change is just removing the `Contributing to the AWS Cloud Development Kit` header which shouldn't be a blocker to getting this merged. can make that change in a separate PR if necessary.

[the remark this answers is not in the document]
Comment (comcalvi) [reply CONTRIBUTING.md:68]: It also expands header into additional subheaders that feel to granular as well, eg

```  
- [Getting Started](#getting-started)
```

goes to

```
- [Getting Started](#getting-started)
    - [Setup](#setup)
    - [Repo Layout](#repo-layout)
    - [Build](#build)
    - [Pack](#pack)
```

@madeline-k and I discussed this offline, the expansions of each header were from a tool that thought it was being helpful and were unintentional changes.

[the remark this answers is not in the document]
Comment (TheRealAmazonKendra) [reply CONTRIBUTING.md:68]: I'm going to go ahead and commit this suggested change so we can get this into the new release.

**Pull request merged**